source/use-cases/sensitive-data-encryption.txt

```diff
@@ -124,7 +126,49 @@ In this step, we generate a local master key to expedite setup of our developmen
           }
 
 
-B. Define a JSON Schema
+.. _fle-create-a-data-encryption-key:
+
+B. Create a Data Encryption Key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this section, we configure the MongoDB driver generate a data
+encryption key using the master key. The data encryption key, used to 
+automatically encrypt and decrypt fields in a document, is stored in a MongoDB 
+collection called the *key vault collection*.
+
+The following configuration values are required by the client to generate a new
+data encryption key:
+
+* The local master key or AWS KMS master key access settings.
+* A MongoDB connection string that authenticates on a running server.
+* The key vault namespace (database and collection).
+
+In this guide, we use the ``encryption`` database and ``__keyvault`` 
+collection as our key vault collection.
+
+The MongoDB driver performs the following to generate the data encryption key:
+
+1.  Creates a data encryption key.
+#.  Inserts a new document into the key vault collection.
```

Comment:
What's in the document? Looks like we give an example document in the final substep - maybe we can elaborate on what's in the document there?

Reply:
Thoughts on moving this process into the content of step three? It's most relevant there IMO.

Reply:
I'm fine with moving it to step three, I was just trying to centralize "pre-" and "post-conditions" in the initial section. Would it also make sense to then move "The client requires the following..." on line 139 to steps 2 and 3 when they are used in the code examples?

Reply:
Delete this list